sources/src/resources/init-scripts/gradle-actions.build-result-capture-service.plugin.groovy

```diff
@@ -41,7 +41,8 @@ abstract class BuildResultsRecorder implements BuildService<BuildResultsRecorder
             // Got EVALUATE SETTINGS event: not a config-cache hit"
             configCacheHit = false
         }
-        if (buildOperation.details in RunRootBuildWorkBuildOperationType.Details) {
+        if (buildOperation.metadata == BuildOperationCategory.RUN_WORK ||
```

Comment:
I wonder why only these build operation types are relevant when I see [there are quite a few](https://github.com/gradle/gradle/blob/ba5282b1d73c087a85c84e5d2b4123662e3dfb47/platforms/core-runtime/base-services/src/main/java/org/gradle/internal/operations/BuildOperationCategory.java#L24)? If `finishEvent.failure != null` does it not necessarily indicate a build failure?

Reply:
It's a good point: I can't recall why I didn't originally do a blanket check for `finishEvent.failure != null`. 

Updated in [46042da](https://github.com/gradle/actions/pull/571/commits/46042da9615b1574512efc6b06bf77b0f327660d).